arcgis-ios-sdk-samples/Edit data/Edit and sync features/EditAndSyncFeaturesViewController.swift

```diff
@@ -0,0 +1,280 @@
+            for geodatabaseFeatureTable in self.geodatabase.geodatabaseFeatureTables {
+                geodatabaseFeatureTable.load { [weak self, unowned geodatabaseFeatureTable] (error: Error?) in
+                    if let error = error {
+                        self?.presentAlert(error: error)
+                    } else {
+                        // Create a new feature layer from the table and add it to the map.
+                        let featureLayer = AGSFeatureLayer(featureTable: geodatabaseFeatureTable)
+                        self?.mapView.map?.operationalLayers.add(featureLayer)
+                    }
+                }
+            }
+            self.generateJob = nil
+            self.barButtonItem.isEnabled = false
+            self.instructionsLabel.text = "Tap on a feature"
+            self.mapView.touchDelegate = self
+        }
+    }
+    
+    func geodatabaseDidSync() {
+        self.presentAlert(title: "Geodatabase sync sucessful")
+        self.barButtonItem.isEnabled = false
+        self.instructionsLabel.text = "Tap on a feature"
+    }
+    
+    func generateGeodatabase() {
+        // Hide the unnecessary items.
+        barButtonItem.isEnabled = false
+        extentView.isHidden = true
+        
+        // Get the area outlined by the extent view.
+        areaOfInterest = self.extentViewFrameToEnvelope()
+        
+        geodatabaseSyncTask.defaultGenerateGeodatabaseParameters(withExtent: areaOfInterest) { [weak self] (params: AGSGenerateGeodatabaseParameters?, error: Error?) in
+            if let params = params,
+                let self = self {
+                // Don't include attachments to minimize the geodatabase size.
+                params.returnAttachments = false
+                
+                // Create a temporary file for the geodatabase.
+                let dateFormatter = ISO8601DateFormatter()
+                
+                let documentDirectoryURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
+                let downloadFileURL = documentDirectoryURL
+                    .appendingPathComponent(dateFormatter.string(from: Date()))
+                    .appendingPathExtension("geodatabase")
+               
+                // Request a job to generate the geodatabase.
+                let generateGeodatabaseJob = self.geodatabaseSyncTask.generateJob(with: params, downloadFileURL: downloadFileURL)
+                self.generateJob = generateGeodatabaseJob
+                generateGeodatabaseJob.start(
+                    statusHandler: { (status: AGSJobStatus) in
+                        SVProgressHUD.show(withStatus: status.statusString()) //Show job status.
+                    },
+                    completion: { [weak self] (_, error: Error?) in
+                        SVProgressHUD.dismiss()
+                        
+                        if let error = error {
+                            self?.presentAlert(error: error)
```

Comment:
A sensible approach which is now supported in Swift is to do this:

``` Swift
guard let self = self else { return }
```

From that point on you have a strong non-optional reference to `self`, having confirmed it has not been deallocated.

Reply:
@sbaskaran If we only reference `weak self` once in any given path through the closure, it is fine to just use optional chaining. Otherwise we unwrap it first using the pattern @nixta showed above.

Reply:
@philium I agree. I've given that proposal at another place in the PR. My suggestion here is to be consistent. Optional chaining is fine, I noticed `self!` somewhere, maybe change that?